core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/auth/ModelAuthProvider.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkiverse.langchain4j.runtime.auth;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.enterprise.inject.Instance;
+import jakarta.enterprise.inject.spi.CDI;
+
+import io.quarkiverse.langchain4j.ModelName;
+
+public interface ModelAuthProvider {
+    String getAuthorization(Input input);
```

Comment:
Hi @csotiriou @geoand, only a couple of questions/suggestions for you to consider.

I was thinking, can we expect situations going forward where local models will require a secure token access, and where a token is passed, for example, as a property ?

I guess one can say that if `Input` is null then do not return a `Bearer ` prefix in the response... This is related to the next question...

The other question, so `ModelAuthProvider` is expected to return a complete HTTP `Authorization` header value, including the scheme, `Bearer <token>`, instead of only `<token>`. In #708, moving to `ModelAuthProvider` has an impact of having to update the test filter to include `Bearer` - I don't think it is a breaking change though, since the current `AuthProvider` is not public.

I wonder, should we expect providers returns the token only, and then filters, depending on a context (access to the remote or local model) will either add `Bearer ` or not.

IMHO it might be a bit better, and a little bit simpler for custom auth providers too...

Reply:
I don't think we need to worry about that now, it's very early days to be able to know exactly how systems using auth will evolve.
For now, I think the proposal in this PR is fine.

Reply:
@geoand Sure, Ok then, this PR works well for #708 so it is good to go from my perspective, thanks